src/service.ts

```diff
@@ -240,6 +240,76 @@ export class Service {
       }, {} as any),
       options,
     };
+
+    const result = await service.compile(input);
+
+    for (const file of files) {
+      file.setProblems([]);
+    }
+
+    const resultItemsIterable = {
+      [Symbol.iterator]: function*() {
+        if (!result.items) {
+          return;
+        }
+        for (const name of Object.keys(result.items)) {
+          yield {name, item: result.items[name]};
+        }
+      }
+    };
+
+    for (const { name, item } of resultItemsIterable) {
+      const { fileRef, console } = item;
+      if (!fileRef || !console) {
+        continue;
+      }
+      const file = fileNameMap[fileRef];
+      if (!file) {
+        continue;
+      }
+      const markers = Service.getMarkers(console);
+      if (markers.length > 0) {
+        monaco.editor.setModelMarkers(file.buffer, "compiler", markers);
+        file.setProblems(markers.map(marker => {
+          return Problem.fromMarker(file, marker);
+        }));
+      }
+    }
+
+    if (!result.success) {
+      throw new Error(result.console);
+    }
+
+    const outputFiles: any = {};
+    for (const { name, item } of resultItemsIterable) {
+      const { content } = item;
+      if (content) {
+        outputFiles[name] = content;
+      }
+    }
+    return outputFiles;
+  }
+
+  static async compileFilesRecursive(files: File[], from: Language, to: Language, options = ""): Promise<{ [name: string]: (string|ArrayBuffer); }> {
```

Comment:
Let's remove this method -- user will specify all files in the array. Create an issue to create "find" the files by pattern (that will be feed into the compileFiles)

Reply:
Can we do this later? We really need to get this landed ASAP.

Reply:
compileFiles currently flattens all files. (`Cargo.toml`, `src/lib.rs` will become `Cargo.toml`, `lib.rs`).